File: dcp-solutions/2024-12-26/solution.go
```markdown
package main

import (
	"container/list"
)

type TreeNode struct {
.Val int
	Left *TreeNode
	Right *TreeNode
}

func printBinaryTreeLevelWise(root *TreeNode) {
	var queue *list.List
	queue = list.New()
	if root != nil {
		queue.PushBack(root)
	}

	for queue.Len() > 0 {
		node := queue.Front()
		root = queue.Remove(node)

		if node.Left != nil {
			queue.PushBack(node.Left)
		}
		if node.Right != nil {
			queue.PushBack(node.Right)
		}

		root.Val
	}
}
```

File: dcp-solutions/2024-12-26/solution.php
```php
class TreeNode {
    public $val;
    public $left;
    public $right;

    function __construct($x) {
        $this->val = $x;
        $this->left = null;
        $this->right = null;
    }
}

function printBinaryTreeLevelWise($root) {
    if (!$root) return;

    $queue = array(array($root, 0));
    while (!empty($queue)) {
        list($node, $level) = array_shift($queue);
        echo $node->val . " ";

        if ($level == 0 && $node->left) {
            array_push($queue, array($node->left, 1));
        }
        if ($level == 0 && $node->right) {
            array_push($queue, array($node->right, 1));
        }

        foreach (array($node->left, $node->right) as $child) {
            if ($child) {
                array_push($queue, array($child, $level + 1));
            }
        }
    }
}
```